vllm/utils.py

```diff
@@ -1312,8 +1312,12 @@ def _load_config_file(file_path: str) -> List[str]:
             raise ex
 
         for key, value in config.items():
-            processed_args.append('--' + key)
-            processed_args.append(str(value))
+            if isinstance(value, bool):
+                if value:
+                    processed_args.append('--' + key)
```

Comment:
I think we have some boolean arguments that aren't just store_true. I know you can specify `--trust-remote-code=True` on the CLI so maybe the issue could also be solved by the mismatch of capitalization on true vs True - what do you think?

Reply:
I searched for `store_false` on this file: https://github.com/vllm-project/vllm/blob/main/vllm/engine/arg_utils.py and couldn't find any instances of it. 

However, with a second look, there's two types of boolean flags: 
- `action='store_true'`  eg `--trust-remote-code`
- `action=StoreBoolean` eg `--disable-logprobs-during-spec-decoding`.

In the store_true case, `--trust-remote-code=True` doesn't work eg run `vllm serve facebook/opt-125m --trust-remote-code=True` and it errors with `vllm serve: error: argument --trust-remote-code: ignored explicit argument 'True'`.

In the `StoreBoolean` case,  `--disable-logprobs-during-spec-decoding=True` does work. 

So I think my code needs to add an extra check for these StoreBoolean arguments

Reply:
Updated the PR to handle `StoreBoolean` values. 

- Moved StoreBoolean to vllm/utils.py to avoid circular imports
- Updated the _load_config_from_file to not be static so it can check which arguments are in StoreBoolean case
- Added a check for the StoreBoolean

Reply:
Ah you were right about the difference there, but yes I was referring to the usage of `StoreBoolean`! Thanks for looking into this